lib/github.ml

```diff
@@ -75,10 +77,32 @@ let validate_signature ?signing_key ~headers body =
   | None -> Error "unable to find header x-hub-signature"
   | Some signature -> if is_valid_signature ~secret signature body then Ok () else Error "signatures don't match"
 
+let get_logs ~ctx ~(api : (module Api.Buildkite)) (n : status_notification) =
+  let module Buildkite = (val api) in
+  let ( let* ) = Lwt_result.bind in
+  match n.state with
+  | Success | Pending -> Lwt.return_ok None
+  | Failure | Error ->
+    let* build = Buildkite.get_build ~ctx n in
+    let failed_jobs =
+      List.filter
+        (fun (job : Buildkite_t.job) ->
+          match job.state with
+          | Some Failed -> true
+          | _ -> false)
+        build.jobs
+    in
+    (match failed_jobs with
+    | [] -> Lwt.return_ok None
+    | _ :: _ ->
+      let* log = Buildkite.get_job_log ~ctx (List.hd failed_jobs) in
+      Lwt.return_ok (Some log.content))
```

Comment:
we don't wan to get the logs for every failed job. We only want to get the logs for the failed jobs we will notify to slack. This should be done in `Action.generate_notifications` before calling `Slack.generate_status_notifications` and then you can pass in the logs to that function. 

I think making the `failed_steps` arg a tuple of `failed_step * log` would be the most correct way